.changelog/v0.38.3/bug-fixes/1827-fix-recheck-async.md

```diff
@@ -0,0 +1,2 @@
+- [`mempool`] Fix data race when rechecking with async ABCI client
```

Comment:
@hvanz this should go to `unreleased` in this PR, or am I missing something?

Reply:
Thanks for pointing out. Fixing it in #3264.